pkg/controllers/monitoring/thanos-querier/controller.go

```diff
@@ -41,6 +42,13 @@ type resourceManager struct {
 	client.Client
 	scheme *runtime.Scheme
 	logger logr.Logger
+	thanos map[string]string
+}
+
+// Options allows for controller options to be set
```

Comment:
(suggestion) should we have a config struct instantiated by main.go rather than map[string]string variables which are less explicit IMHO?

````go
type OperandConfiguration struct {
    Image string
    Version string
}

type OperatorConfiguration struct {
    Prometheus OperandConfiguration
    Alertmanager OperandConfiguration
    Thanos OperandConfiguration
}
````

Reply:
Yeah I agree, that would improve readability. I did play around with that too and I didn't go that way because I wasn't sure how to get the key-value style argument parsing paired up with a struct without going through some function involving `reflect`. 

I missed that I didn't add the changes for the `versions` cli arg, added now.

Do you know how I can parse two repeatable cli args into a single struct? Otherwise I wonder if its worth going through the trouble to convert `map[string]string` to a struct.

Reply:
Do we expect this to grow at some point? If so, +1 on the struct but just for image/version I think it may not be worth the change.